miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -243,87 +243,72 @@ end
+    dup.1 push.PUBLIC_NOTE dup.1 dup.3
+    # => [tag, note_type, public_note, note_type, tag, note_type, RECIPIENT]
 
     u32assert.err=ERR_NOTE_TAG_MUST_BE_U32
-    # => [tag, note_type, public_note, note_type, ASSET, tag, note_type, RECIPIENT]
+    # => [tag, note_type, public_note, note_type, tag, note_type, RECIPIENT]
 
     # enforce the note type depending on the tag's bits
     u32shr.30 push.ALL_NOTE_TYPES_ALLOWED eq cdrop assert_eq.err=ERR_NOTE_INVALID_TYPE_FOR_TAG
-    # => [ASSET, tag, note_type, RECIPIENT]
+    # => [tag, note_type, RECIPIENT]
 
     # get the index for the next note to be created and increment counter
     exec.increment_num_created_notes
-    # => [note_idx, ASSET, tag, note_type, RECIPIENT]
+    # => [note_idx, tag, note_type, RECIPIENT]
 
     # get a pointer to the memory address at which the note will be stored
     exec.memory::get_created_note_ptr
-    # => [note_ptr, ASSET, tag, note_type, RECIPIENT]
+    # => [note_ptr, tag, note_type, RECIPIENT]
 
     # populate the metadata
-    movup.5 exec.account::get_id movup.7
-    # => [note_type, sender_acct_id, tag, note_ptr, ASSET, RECIPIENT]
+    swap exec.account::get_id movup.3
+    # => [note_type, sender_acct_id, tag, note_ptr, RECIPIENT]
 
     push.0 # TODO: allow the user to push the aux data
-    # => [aux, note_type, sender_acct_id, tag, note_ptr, ASSET, RECIPIENT]
+    # => [aux, note_type, sender_acct_id, tag, note_ptr, RECIPIENT]
 
     emit.NEW_NOTE_EVENT
 
     # set the metadata for the new created note
     dup.4 exec.memory::set_created_note_metadata
-    # => [note_ptr, ASSET, RECIPIENT]
-
-    # set the number of assets for the note to 1
-    push.1 dup.1 exec.memory::set_created_note_num_assets
-    # => [note_ptr, ASSET, RECIPIENT]
-
-    movdn.4 padw swapw movup.8
-    # => [note_ptr, ASSET, 0, 0, 0, 0, RECIPIENT]
-
-    # add the asset to the note
-    dup movdn.5 exec.memory::get_created_note_asset_data_ptr mem_storew dropw
-    # => [note_ptr, 0, 0, 0, 0, RECIPIENT]
-
-    movdn.8 swapw padw swapw movup.12
-    # => [note_ptr, RECIPIENT, 0, 0, 0, 0, 0, 0, 0, 0]
+    # => [note_ptr, RECIPIENT]
 
     # set the recipient
     dup movdn.5 exec.memory::set_created_note_recipient
-    # => [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0]
+    # => [note_ptr]
 end
 
 #! Adds an ASSET to the specified note and returns a pointer to the memory address
 #! at which the note is stored.
 #!
-#! Panics if
-#! - the ASSET is malformed (invalid faucet ID)
-#! - the max amount of fungible assets is exceeded
-#! - the non-fungible asset already exists in the note
-#!
 #! Inputs: [note_ptr, ASSET]
 #! Outputs: [note_ptr]
 #!
 #! note_ptr is the pointer to the memory address at which the note is stored.
 #! ASSET can be a fungible or non-fungible asset.
+#!
+#! Panics if
+#! - the ASSET is malformed (invalid faucet ID)
+#! - the max amount of fungible assets is exceeded
+#! - the non-fungible asset already exists in the note
```

Comment:
There is one failure condition missing: we are tying to add more assets than is allowed per note (255, I believe).